### PR TITLE
[TASK] Allow renaming dashboards

### PR DESCRIPTION
**Pull request by KoenWouters**
In add dashboard modal there is an option to give the new dashboard a custom name. Also added a config option on top to config current dashboard (for now only rename current dashboard).

### CODE REVIEW

**Comment by KoenWouters** [comment]
When editing a default dashboard the label string is in the input field instead of the "default" text

File: Resources/Private/Templates/Dashboard/Main.html [hunk 1]
```diff
@@ -103,31 +112,60 @@ <h3><f:translate key="dashboard.empty.content.title" extensionName="dashboard" /
...
-                            </div>
-                            <f:if condition="{availableDashboardConfig.description}">
-                                <div class="dashboardModal-item-description">
-                                    <f:translate key="{availableDashboardConfig.description}" default="{availableDashboardConfig.description}"/>
-                                </div>
-                            </f:if>
+<div id="dashboardModal-add" class="hide">
+    <f:form actionUri="{addDashboardUri}" class="dashboardModal-form" target="list_frame">
+            <div class="dashboardModal-formInput form-group">
+                <label for="dashboardModal-title">Dashboard Title</label>
```
Comment (haassie) [inline Resources/Private/Templates/Dashboard/Main.html]: Hardcoded labels?

File: Resources/Private/Templates/Dashboard/Main.html [hunk 2]
```diff
@@ -103,31 +112,60 @@ <h3><f:translate key="dashboard.empty.content.title" extensionName="dashboard" /
...
-                    </f:be.link>
-                </div>
-            </f:if>
-        </f:for>
+                    </f:if>
+                </f:for>
+            </div>
+
+            <div class="dashboardModal-submit">
+                <f:form.submit value="Create Dashboard" class="dashboardModal-button"/>
```
Comment (haassie) [inline Resources/Private/Templates/Dashboard/Main.html]: Hardcoded label

File: Resources/Private/Templates/Dashboard/Main.html [hunk 3]
```diff
@@ -103,31 +112,60 @@ <h3><f:translate key="dashboard.empty.content.title" extensionName="dashboard" /
...
+
+        </f:form>
+</div>
+
+<div id="dashboardModal-configure" class="hide">
+    <div class="dashboard-modal-items">
+        <f:form actionUri="{configureDashboardUri}" class="dashboardModal-form" target="list_frame">
+            <f:form.hidden name="currentDashboard" value="{currentDashboard}" />
+            <div class="dashboardModal-formInput form-group">
+                <label for="dashboardModal-title">Dashboard Title</label>
```
Comment (haassie) [inline Resources/Private/Templates/Dashboard/Main.html]: Same

File: Resources/Private/Templates/Dashboard/Main.html [hunk 4]
```diff
@@ -103,31 +112,60 @@ <h3><f:translate key="dashboard.empty.content.title" extensionName="dashboard" /
...
+                <f:form.textfield name="dashboard[label]"  id="dashboardModal-title" class="form-control" value="{currentDashboardName}" />
+            </div>
+
+            <div class="dashboardModal-submit">
+                <f:form.submit value="Save Dashboard" class="dashboardModal-button"/>
```
Comment (haassie) [inline Resources/Private/Templates/Dashboard/Main.html]: Same

File: Resources/Private/Templates/Dashboard/Main.html [hunk 5]
```diff
@@ -8,14 +8,23 @@ <h1>
...
+        <a href="{addDashboardUri}" class="js-dashboard-modal dashboardControl" data-modal-identifier="add" data-modal-title="{f:translate(key: 'dashboard.add', extensionName: 'dashboard')}" data-button-close-text="{f:translate(key: 'dashboard.add.button.close', extensionName: 'dashboard')}" data-button-ok-text="{f:translate(key: 'dashboard.add.button.ok', extensionName: 'dashboard')}">
             <core:icon identifier="actions-add" alternativeMarkupIdentifier="inline" /><span class="sr-only"><f:translate key="dashboard.add" extensionName="dashboard"/></span>
         </a>
     </div>
+
+    <div class="dashboardConfiguration">
+        <a href="{configureDashboardUri}" class="js-dashboard-modal" data-modal-identifier="configure" data-modal-title="{f:translate(key: 'dashboard.configure', extensionName: 'dashboard')}" data-button-close-text="{f:translate(key: 'dashboard.add.button.close', extensionName: 'dashboard')}" data-button-ok-text="{f:translate(key: 'dashboard.add.button.ok', extensionName: 'dashboard')}">
```
Comment (haassie) [inline Resources/Private/Templates/Dashboard/Main.html]: The language label `dashboard.configure` is missing

File: Tests/Functional/Dashboards/DashboardRepositoryTest.php [hunk 6]
```diff
@@ -100,11 +100,51 @@ public function getDashboardByIdentifierReturnsArrayOneDashboard(): void
...
+     * @throws \TYPO3\TestingFramework\Core\Exception
+     * @test
+     */
+    public function updateDashboardSettingsChangesDashboardLabel(): void
+    {
+        $this->importDataSet(__DIR__ . '/../Fixtures/sys_dashboards_one_dashboard.xml');
+        $dashboardLabel = 'Renamed Dashboard';
+
+        $this->subject->updateDashboardSettings(
+            'a8a9ad23c27c51640738fcae687563243af5a58f',
```
Comment (haassie) [inline Tests/Functional/Dashboards/DashboardRepositoryTest.php]: Maybe move this identifier to a separate variable?

**Pull request merged**